frontend/src/scenes/experiments/experimentLogic.tsx

```diff
@@ -902,6 +903,12 @@ export const experimentLogic = kea<experimentLogicType>([
             }))
 
             const newMetricsIds = sharedMetricIds.map((id: SharedMetric['id']) => ({ id, metadata }))
+            newMetricsIds.forEach((metricId) => {
+                const metric = values.sharedMetrics.find((m: SharedMetric) => m.id === metricId.id)
+                if (metric) {
+                    actions.reportExperimentSharedMetricAssigned(values.experimentId, metric)
```

Comment:
Not a strong opinion, but would it make sense to make this action take in a list of shared metrics instead of having getting one event for each metric? as they are added in one action?

Reply:
@andehen I thought about that. I think the advantage of firing individual events is that it makes event properties less complex/easier to compare/etc. What do you think is the advantage of sending a list of shared metrics?

Reply:
I think perhaps it would be cleaner from a data perspective in that one user action leads to one event. So answering questions like how many times this action is performed, and what the typical number of shared metrics added is would perhaps be easier. But not sure, just thinking out loudly here.